### PR TITLE
[Gutenberg] Voice to content

### PR DESCRIPTION
**Pull request by geriux**
**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6878
- https://github.com/WordPress/gutenberg/pull/61796

To test:

A [hard-coded call](https://github.com/wordpress-mobile/WordPress-iOS/pull/23234/files#diff-beacfffc3f8a87edf94895da980a81e356f1b5aa7998f70791bc7baec9a11731) has been added to test the integration with the editor.

- Install the build from this RP
- Open the editor
- Expect the content to be added once the editor opens.

## Regression Notes
1. Potential unintended areas of impact

It should only affect the Editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing

3. What automated tests I added (or what prevented me from doing so)

Integration tests were added in Gutenberg Mobile

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift [hunk 1]
```diff
@@ -884,6 +884,8 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
             insertPrePopulatedMedia()
             focusTitleIfNeeded()
             mediaInserterHelper.refreshMediaStatus()
+
+            gutenberg.onVoiceToContent(content: "# Sample Document\nLorem ipsum dolor sit amet, consectetur adipiscing elit.\n## Overview\n- Lorem ipsum dolor sit amet\n- Consectetur adipiscing elit\n- Integer nec odio\n## Details\n1. Sed cursus ante dapibus diam\n2. Nulla quis sem at nibh elementum imperdiet\n3. Duis sagittis ipsum\n## Mixed Lists\n- Key Points:\n  1. Lorem ipsum dolor sit amet\n  2. Consectetur adipiscing elit\n  3. Integer nec odio\n- Additional Info:\n  - Sed cursus ante dapibus diam\n  - Nulla quis sem at nibh elementum imperdiet\n  - Duis sagittis ipsum\n## Conclusion\n- Praesent libero\n- Sed cursus ante dapibus diam\n- Suspendisse malesuada lacus ex")
```
Comment (geriux) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift]: This will be removed when the PR is approved and ready to be merged

**Comment by geriux** [comment]
Tests fail because of the content added into the editor. Those should pass once we remove the hard-coded call to `onVoiceToContent`.

**Review by twstokes — APPROVED** [review]
LGTM @geriux! 🚀

**Comment by kean** [comment]
Hey, @geriux. I've been testing, and it's been working great so far, but I found a couple of missing features:

- It correctly converts `# Title` to `<h1>Title</h1>` but doesn't place it in the dedicated space for the post title
- If the original text contains _only_ the title, it doesn't seem to produce any output

**Comment by geriux** [comment]
> Hey, @geriux. I've been testing, and it's been working great so far, but I found a couple of missing features:
> 
> * It correctly converts `# Title` to `<h1>Title</h1>` but doesn't place it in the dedicated space for the post title
> * If the original text contains _only_ the title, it doesn't seem to produce any output

Thank you for the feedback @kean! I'll take a look!

File: WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 2]
```diff
@@ -1,78 +1,6 @@
 {
   "object": {
     "pins": [
-      {
```
Comment (geriux) [inline WordPress.xcworkspace/xcshareddata/swiftpm/Package.resolved]: I'll revert this change, not sure why it got modified

**Comment by twstokes** [comment]
> * It correctly converts `# Title` to `<h1>Title</h1>` but doesn't place it in the dedicated space for the post title

➕ I should've called that out, but assumed it was expected in the test. I saw it on both Android and iOS. Thanks @kean.

**Pull request merged**